src/components/ApplicationLauncher/ApplicationLauncher.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import ApplicationLauncherToggle from './ApplicationLauncherToggle';
+import { Dropdown } from '../Dropdown';
+
+class ApplicationLauncher extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showLauncher: false,
+      type: props.type
+    };
+  }
+
+  componentWillReceiveProps = nextProps => {
+    this.setState({ type: nextProps.type });
+  };
+
+  toggleLauncher = () => {
+    this.setState({ showLauncher: !this.state.showLauncher });
+  };
+
+  render() {
+    const classes = classNames(
+      {
+        'applauncher-pf applauncher-pf-block-list dropdown dropdown-kebab-pf':
+          this.state.type === 'grid'
+      },
+      {
+        'applauncher-pf dropdown dropdown-kebab-pf': this.state.type === 'list'
+      },
+      { open: this.state.showLauncher }
```

Comment:
Would be better to always specify `applauncher-pf dropdown dropdown-kebab-pf` and add `applauncher-pf-block-list` if type === grid.